core/src/main/java/com/linecorp/armeria/client/ClientRequestContext.java

```diff
@@ -322,24 +323,25 @@ static ClientRequestContextBuilder builder(RpcRequest request, URI uri) {
      */
     @Override
     default SafeCloseable push() {
-        final RequestContext oldCtx = RequestContextThreadLocal.getAndSet(this);
+        final ContextStorage contextStorage = RequestContextUtil.storage();
```

Comment:
Not sure if it's better but `RequestContextUtil` could still expose the same static API as before without returning `ContextStorage`